cmds/dhclient/dhcp6client/dhcp6client.go

```diff
@@ -0,0 +1,138 @@
+package dhcp6client
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net"
+	"time"
+
+	"github.com/mdlayher/dhcp6"
+	"github.com/vishvananda/netlink"
+)
+
+type Client struct {
+	hardwareAddr  net.HardwareAddr //The HardwareAddr to send in the request.
+	ignoreServers []net.IP         //List of Servers to Ignore requests from.
+	timeout       time.Duration    //Time before we timeout.
+	broadcast     bool             //Set the Bcast flag in BOOTP Flags
+	connection    connection       //The Connection Method to use
+}
+
+/*
+*  * Abstracts the type of underlying socket used
+*   */
+type connection interface {
+	Close() error
+	Write(packet []byte) error
+	ReadFrom()
+	// SetReadTimeout(t time.Duration) error
+}
+
+func New(haddr net.HardwareAddr, conn connection, timeout time.Duration) (*Client, error) {
```

Comment:
It seems like you are trying to both refactor and add a new feature. Let's do one at a time first? If you just copy dhclient4 and make it dhclient6, and then focus on getting a dhcp6client alternative, get that working, *then* you can get rid of duplicated code by refactoring.